src/jl_uv.c

```diff
@@ -1085,6 +1085,25 @@ JL_DLLEXPORT int jl_queue_work(work_cb_t work_func, void *work_args, void *work_
     return 0;
 }
 
+JL_DLLEXPORT int jl_uv_update_timer_start(uv_loop_t* loop, uv_timer_t* handle,
+                                          uv_timer_cb cb, uint64_t timeout,
+                                          uint64_t repeat)
+{
+    JL_UV_LOCK();
+    int err = uv_timer_init(loop, handle);
+    if (err) {
+        // TODO: this codepath is currently not tested
+        free(handle);
+        return err;
+    }
+
+    jl_uv_associate_julia_struct((uv_handle_t *)handle, (jl_value_t *)handle);
```

Comment:
one of these things is not like the other—you can't cast to both libuv and julia

Reply:
Sorry, I'm getting slightly confused by what's what, since in the following two locations it seems that we're already passing objects which (to my uneducated eyes) seem to end up with casts that are essentially the same as what you pointed out at:

https://github.com/JuliaLang/julia/blob/4b0c8e7edb848e8f8b9047ee12678d8e8b0609bf/base/asyncevent.jl#L90
https://github.com/JuliaLang/julia/blob/4b0c8e7edb848e8f8b9047ee12678d8e8b0609bf/base/asyncevent.jl#L102-L103

Those two libuv calls don't expect a `jl_value_t*`, even though that's what we're doing there, right? I assume it works because `this.handle` is the first field in the `Timer` struct so the pointers are the same anyway?